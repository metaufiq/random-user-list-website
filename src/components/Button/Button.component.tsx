import { Props } from "./Button.component.types";

const Button = (props: Props)=>{
  return(
    <button 
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline" 
      type="button"
      onClick={props.onClick}>
      {props.label}
    </button>
  )
}

export default Button;