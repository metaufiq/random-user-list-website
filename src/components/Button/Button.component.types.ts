export interface Props{
  label: string,
  onClick?: VoidFunction
}