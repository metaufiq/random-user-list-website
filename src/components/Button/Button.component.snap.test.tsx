import { assertSnapshots } from "../../../test";
import Button from ".";
import { Props } from "./Button.component.types";

describe('ButtonSnapshot', () => {
  const mockProps:Props = {
    children: 'children'
  };

  const configs = [
    {
      props: {...mockProps, onClick: ()=>{}},
      desc: 'should render normal Button'
    },
    {
      props: mockProps,
      desc: 'should render Button without onClick'
    }
  ];

  assertSnapshots(Button, configs);
});
