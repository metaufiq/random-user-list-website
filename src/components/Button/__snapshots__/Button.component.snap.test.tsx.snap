// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSnapshot should render Button without onClick 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
  type="button"
>
  label
</button>
`;

exports[`ButtonSnapshot should render normal Button 1`] = `
<button
  className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
  onClick={[Function]}
  type="button"
>
  label
</button>
`;
